drivers/frequency/adf4368/iio_adf4368.c

```diff
@@ -0,0 +1,1199 @@
+	int ret;
+
+	if (!iio_adf4368)
+		return -EINVAL;
+
+	adf4368 = iio_adf4368->adf4368_dev;
+
+	if (!adf4368)
+		return -EINVAL;
+
+	ret = adf4368_get_phase(adf4368, &phase, &phase_pol);
+	if (ret)
+		return ret;
+
+	val = phase_pol ? -phase : phase;
+	return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+}
+
+/**
+ * @brief Handles the write request for the phase adjustment attribute.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with the data to be written.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return 	  - Result of the writing procedure, error code otherwise.
+ */
+static int adf4368_iio_write_phase(void *dev, char *buf, uint32_t len,
+				   const struct iio_ch_info *channel,
+				   intptr_t priv)
+{
+	struct adf4368_iio_dev *iio_adf4368 = (struct adf4368_iio_dev *)dev;
+	struct adf4368_dev *adf4368;
+	bool phase_pol;
+	int32_t val;
+
+	if (!iio_adf4368)
+		return -EINVAL;
+
+	adf4368 = iio_adf4368->adf4368_dev;
+
+	if (!adf4368)
+		return -EINVAL;
+
+	iio_parse_value(buf, IIO_VAL_INT, &val, NULL);
+	phase_pol = val < 0 ? true : false;
+
+	val = abs(val);
+
+	return adf4368_set_phase(adf4368, val, phase_pol);
+
+}
+
+
+
+static struct iio_attribute adf4368_iio_attrs[] = {
+	{
+		.name = "bleed_current",
+		.shared = IIO_SHARED_BY_ALL,
+		.show = adf4368_iio_read_bleed_i,
+		.store = adf4368_iio_write_bleed_i,
+	},
+	{
+		.name = "charge_pump_current",
+		.shared = IIO_SHARED_BY_ALL,
+		.show = adf4368_iio_read_cpi,
+		.store = adf4368_iio_write_cpi,
+	},
+	{
+		.name = "charge_pump_current_available",
+		.shared = IIO_SHARED_BY_ALL,
+		.show = adf4368_iio_cpi_available,
+	},
+	{
+		.name = "default_regs",
+		.shared = IIO_SHARED_BY_ALL,
+		.show = adf4368_iio_read_default_regs,
+		.store = adf4368_iio_write_default_regs,
+	},
+	{
+		.name = "reference_doubler_en",
+		.shared = IIO_SHARED_BY_ALL,
+		.show = adf4368_iio_read_rdoubler,
+		.store = adf4368_iio_write_rdoubler,
+	},
+	{
+		.name = "reference_divider",
+		.shared = IIO_SHARED_BY_ALL,
+		.show = adf4368_iio_read_rdivider,
+		.store = adf4368_iio_write_rdivider,
+	},
+	{
+		.name = "reference_frequency",
+		.shared = IIO_SHARED_BY_ALL,
+		.show = adf4368_iio_read_ref_freq,
+		.store = adf4368_iio_write_ref_freq,
+	},
+	{
+		.name = "sync_setup",
+		.shared = IIO_SHARED_BY_ALL,
+		.show = adf4368_iio_read_sync_setup,
+		.store = adf4368_iio_write_sync_setup,
+	},
+	{
+		.name = "sw_sync",
+		.shared = IIO_SHARED_BY_ALL,
+		.show = adf4368_iio_read_sw_sync,
+		.store = adf4368_iio_write_sw_sync,
+	},
+	{
+		.name = "temperature",
+		.shared = IIO_SHARED_BY_ALL,
+		.show = adf4368_iio_read_temperature,
+		.store = adf4368_iio_write_temperature,
+	},
+	END_ATTRIBUTES_ARRAY
+};
+
+
+
+static struct iio_attribute adf4368_iio_ch_attrs[] = {
+	{
+		.name = "en",
+		.shared = IIO_SEPARATE,
+		.show = adf4368_iio_read_en,
```

Comment:
instead of doing 4 different read and 4 different write functions for these attributes you could've done 1 single read and 1 write function

please see how it was done here:
https://github.com/analogdevicesinc/no-OS/blob/cc73a3d70d056d143b783bd846489e2d2819ba7c/drivers/rf-transceiver/hmc630x/iio_hmc630x.c#L326-L366

i know there are lots of places that define a function for an attribute (i've done it too) but we should not be doing it like that anymore, it just takes 200 lines of code to write something that could be done in 50

Reply:
Fixed and tested